### DISCUSSION TITLE
Why is Istio released with a pre-release (dev) version of envoy?

### DISCUSSION THREAD

**Discussion by Oberon00** [istio/istio#48064] [Q&A] [answered]
For example, Istio 1.20.0 deploys envoy 1.28.1-dev in sidecars (the latest release of envoy at this time is 1.28.0). I also have checked an old Istio 1.16.2 installation which seems to come with envoy 1.24.2-dev.  Why are pre-releases used instead of the latest release? How is the envoy version / exact commit selected?

**Comment by zirain** [comment]
find the related envoy commit from [here](https://github.com/istio/proxy/blob/b2c85adf00d6b1409d37c93dc37f52d22237045e/WORKSPACE#L25-L32)

**Reply by Oberon00** [reply]
It seems this is updated more often than envoy releases new versions. Why does Istio not simply use (or base its adaptions on) the latest public release of envoy?

**Reply by howardjohn** [reply, marked as answer]
We build and manage our own builds of envoy and are decoupled form their official patch release versions intentionally. The patch version from envoy is effectively an arbitrary decision to trigger a release; we make that decision ourselves instead of relying on Envoy, largely to get updates we need faster.

Note this is only patches, which are pretty infrequent. We do follow minor releases and release branches

**Reply by parshva-shah-dsw** [reply]
@howardjohn can we use envoy version with -dev tag in production environments which is bundled in istio 1.19.0+ versions
It also appears that -dev tag is missing with all the supported envoy release document https://istio.io/latest/docs/releases/supported-releases/#supported-envoy-versions

**Reply by howardjohn** [reply]
As an Istio user you don't pick the envoy version. We do as Istio maintainers. And the version we pick are intended for production use.
